plugins/Goals/PagesBeforeCalculator.php

```diff
@@ -92,58 +91,49 @@ public function calculateFor(?string $startDatetime, ?string $endDatetime, ?int
                 // All goals
                 $goalsModel = new GoalsModel();
                 $goals = array_column($goalsModel->getActiveGoals([$site]), 'idgoal');
+                $goals[] = 0; // Include ecommerce orders
```

Comment:
We should avoid processing the ecommerce goal if its not enabled.
```suggestion
                if (\Piwik\Site::isEcommerceEnabledFor($site) {
                    $goals[] = \Piwik\Tracker\GoalManager::IDGOAL_ORDER;
                }
```